🐛 Test environment contamination causes auth test failures

## Description
The test suite has an environment contamination issue where environment variables from one test persist and affect subsequent tests, causing intermittent test failures.

## Failing Test
- `tests/test_auth.py::TestAuthManager::test_load_credentials_incomplete`

## Root Cause
1. The `test_default_config_path` test creates an `AuthManager()` without specifying a config path
2. This uses the default config path (`~/.config/youtrack-cli/.env`) 
3. The `AuthManager.__init__` method calls `load_dotenv(self.config_path)`, loading environment variables from the default config file into the process environment
4. The default config file contains `YOUTRACK_TOKEN=perm:cnlhbg==.NDYtMA==.OS0q829lf7zZDdqtDvMaZ6jrR6jAyJ`
5. This token persists in the environment for the duration of the test process
6. When `test_load_credentials_incomplete` runs, it patches the environment to only include `YOUTRACK_BASE_URL`, but the `YOUTRACK_TOKEN` from the previous test is still present
7. The test expects `load_credentials()` to return `None` (incomplete credentials), but it returns a valid `AuthConfig` because both base URL and token are available

## Expected Behavior
Tests should be isolated and not affect each other's environment state.

## Current Failure
```
AssertionError: assert AuthConfig(base_url='https://example.youtrack.cloud', token='perm:cnlhbg==.NDYtMA==.OS0q829lf7zZDdqtDvMaZ6jrR6jAyJ', username=None) is None
```

## Reproduction Steps
1. Run the full test suite: `uv run python -m pytest tests/test_auth.py`
2. Observe that `test_load_credentials_incomplete` fails
3. Run the same test in isolation: `uv run python -m pytest tests/test_auth.py::TestAuthManager::test_load_credentials_incomplete -xvs`
4. Observe that it passes when run alone

## Possible Solutions
1. **Test Isolation**: Use `pytest` fixtures or setup/teardown methods to clear environment variables between tests
2. **Mock dotenv loading**: Mock the `load_dotenv` calls in tests to prevent loading from actual config files
3. **Environment restoration**: Save and restore the environment state in test setup/teardown
4. **Test ordering**: Ensure tests that modify environment run in isolation or last

## Priority
Medium - This causes intermittent test failures that can block CI/CD pipelines and development workflow.

## Files Affected
- `tests/test_auth.py` - Contains the failing test and the test causing contamination
- `youtrack_cli/auth.py` - Contains the AuthManager class that loads environment variables
- `~/.config/youtrack-cli/.env` - Default config file that gets loaded during tests

## Impact
- Intermittent CI/CD failures
- Developer confusion when tests pass/fail inconsistently
- Potential masking of real authentication bugs due to unexpected environment state

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,0 +1,40 @@
+"""Global test configuration and fixtures."""
+
+import os
+from typing import Optional
+from unittest.mock import patch
+
+import pytest
+
+
+@pytest.fixture(scope="function", autouse=True)
+def isolate_environment():
+    """Ensure test isolation by saving and restoring environment variables."""
+    # YouTrack environment variables that could cause test contamination
+    youtrack_env_vars = ["YOUTRACK_BASE_URL", "YOUTRACK_TOKEN", "YOUTRACK_USERNAME"]
+
+    # Store original values
+    original_env: dict[str, Optional[str]] = {}
+    for key in youtrack_env_vars:
+        original_env[key] = os.environ.get(key)
+
+    # Clear YouTrack environment variables before each test
+    for key in youtrack_env_vars:
+        if key in os.environ:
+            del os.environ[key]
+
+    yield
+
+    # Restore original environment after test
+    for key, value in original_env.items():
+        if value is not None:
+            os.environ[key] = value
+        elif key in os.environ:
+            del os.environ[key]
+
+
+@pytest.fixture(scope="function", autouse=True)
+def mock_dotenv_loading():
+    """Mock dotenv loading to prevent real config files from loading."""
+    with patch("youtrack_cli.auth.load_dotenv") as mock_load_dotenv:
+        yield mock_load_dotenv

--- a/tests/test_auth.py
+++ b/tests/test_auth.py
@@ -49,6 +49,24 @@ class TestAuthManager:
         self.config_path = os.path.join(self.temp_dir, ".env")
         self.auth_manager = AuthManager(self.config_path)
 
+        # Store original environment variables to restore later
+        self.original_env = {
+            key: os.environ.get(key)
+            for key in ["YOUTRACK_BASE_URL", "YOUTRACK_TOKEN", "YOUTRACK_USERNAME"]
+        }
+
+    def teardown_method(self):
+        """Clean up test fixtures and restore environment."""
+        # Remove any YouTrack environment variables that were set during tests
+        for key in ["YOUTRACK_BASE_URL", "YOUTRACK_TOKEN", "YOUTRACK_USERNAME"]:
+            if key in os.environ:
+                del os.environ[key]
+
+        # Restore original environment variables
+        for key, value in self.original_env.items():
+            if value is not None:
+                os.environ[key] = value
+
     def test_default_config_path(self):
         """Test default config path generation."""
         manager = AuthManager()

